pkg/sql/codegen/common.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2020 The SQLFlow Authors. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package codegen
+
+import (
+	"fmt"
+	"sqlflow.org/sqlflow/pkg/ir"
+	"sqlflow.org/sqlflow/pkg/util"
+	"strings"
+)
+
+func toModuleDataType(dtype int, module string) (string, error) {
+	switch dtype {
+	case ir.Int:
+		return fmt.Sprintf("%s.dtypes.int64", module), nil
+	case ir.Float:
+		return fmt.Sprintf("%s.dtypes.float32", module), nil
+	case ir.String:
+		return fmt.Sprintf("%s.dtypes.string", module), nil
+	default:
+		return "", fmt.Errorf("unsupport dtype: %d", dtype)
+	}
+}
+
+// TODO(sneaxiy): XGBoost does not support some feature columns, such as EMBEDDING.
+// For better error message, we should find a better way to distinguish whether the
+// module is TensorFlow or XGBoost.
+func isTensorFlowModule(module string) bool {
```

Comment:
In the future, we will support PyTorch.
We cannot assume that !isTensorFlowModule equals to isXGBoostModule

Reply:
Thanks! Renamed `isTensorFlowModule` to be `isXGBoostModule` in https://github.com/sql-machine-learning/sqlflow/pull/2402 .